src/api/LiikuntaApolloClient.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  ApolloClient,
+  ApolloQueryResult,
+  DocumentNode,
+  gql,
+  NormalizedCacheObject,
+  OperationVariables,
+  QueryOptions,
+} from "@apollo/client";
+import { OperationDefinitionNode, SelectionNode } from "graphql";
+import { GetStaticPropsContext } from "next";
+
+import { Language, MenuItem } from "../types";
+import mockMenuItems from "./tmp/menuItems";
+
+type GlobalData = {
+  globalLanguages: Language[];
+  globalMenuItems: { nodes: MenuItem[] };
+};
+
+function getSelectionSet(documentNode: DocumentNode): SelectionNode[] {
+  return documentNode.definitions.find(
+    (node): node is OperationDefinitionNode =>
+      node.kind === "OperationDefinition"
+  )?.selectionSet?.selections as SelectionNode[];
+}
+
+const GLOBAL_QUERY = gql`
+  {
+    globalLanguages: languages {
+      id
+      name
+      slug
+      code
+      locale
+    }
```

Comment:
I think in all the other front-end applications the graphql queries & mutations are in their own files. Do we wan't to make same structure here or keep them as they are now?

Reply:
I think that this particular query is is more of a temporary hack than something that should remain for the long term--in that sense I think that this should be managed within this file so that it's easy to remove later.

As for the other queries, I  don't have as clear of an opinion. Apollo recommends storing queries alongside components. For the page components, the only place is the file itself. For other components, it's possible to store the queries alongside the component file in a separate file.

I don't really have a strong opinion. By default I would maybe prefer the more simple approach which I my opinion is to not create a separate file What pros and cons do you see? Do you prefer one of the approaches?

Reply:
I'll merge this PR, but we can continue the discussion. Let's make another PR to fix this if we decide to. I have other changes that are awaiting that these commits land on main.